python/hail/ir/table_ir.py

```diff
@@ -235,14 +235,19 @@ def __init__(self, child, n, shuffle):
         self.n = n
         self.shuffle = shuffle
 
-    def __str__(self):
-        return f'(TableRepartition {self.n} {self.shuffle} {self.child})'
+    def render(self, r):
+        return f'(TableRepartition {self.n} {self.shuffle} {r(self.child)})'
 
 class LocalizeEntries(TableIR):
     def __init__(self, child, entry_field_name):
         super().__init__()
         self.child = child
         self.entry_field_name = entry_field_name
 
-    def __str__(self):
-        return f'(LocalizeEntries "{escape_str(self.entry_field_name)}" {self.child})'
+    def render(self, r):
+        return f'(LocalizeEntries "{escape_str(self.entry_field_name)}" {r(self.child)})'
+
+class JavaTable(TableIR):
+    def __init__(self, jir):
+        self._jir = jir
+    # render handled by Renderer
```

Comment:
why not handle here?

Reply:
I thought it would always be handled by render, but that's not true if `stop_at_jir` is false, so I added it.